tests/cpp/core/ScreenLockTests.cpp

```diff
@@ -0,0 +1,109 @@
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#define BOOST_TEST_MODULE ScreenLockTest
+
+#include <boost/test/unit_test.hpp>
+
+#include "ScreenLock.h"
+
+#include <QObject>
+
+namespace
+{
+const QString STREAM1("Stream1");
+const QString STREAM2("Stream2");
+}
+
+BOOST_AUTO_TEST_CASE(test_locking)
+{
+    auto screenLock = ScreenLock::create();
+    bool locked = false;
+    bool listChanged = false;
+    bool lockModified = false;
+    bool streamAccepted = false;
+    bool streamRejected = false;
+
+    QObject::connect(screenLock.get(), &ScreenLock::lockChanged,
+                     [&locked](bool lockState) { locked = lockState; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::streamListChanged,
+                     [&listChanged]() { listChanged = true; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::modified,
+                     [&lockModified]() { lockModified = true; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::streamAccepted,
+                     [&streamAccepted](QString) { streamAccepted = true; });
+
+    QObject::connect(screenLock.get(), &ScreenLock::streamRejected,
+                     [&streamRejected](QString) { streamRejected = true; });
+
+    screenLock->lock();
+    BOOST_CHECK_EQUAL(locked, true);
+    BOOST_CHECK_EQUAL(lockModified, true);
+
+    screenLock->requestStreamAcceptance(STREAM1);
+    BOOST_CHECK_EQUAL(listChanged, true);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 1);
+
+    screenLock->acceptStream(STREAM1);
+    BOOST_CHECK_EQUAL(streamAccepted, true);
+    BOOST_CHECK(screenLock->getPendingStreams().length() == 0);
```

Comment:
please also check that calling acceptStream(STREAM1); a second time does not emit streamAccepted.